CHANGELOG.md

```diff
@@ -3,6 +3,7 @@
 ### Enhancements
 * UUID allowed as partition value ([#4500](https://github.com/realm/realm-core/issues/4500))
 * The error message when the intial steps of opening a Realm file fails is now more descriptive.
+* If a string with invalid size is encountered during file upgrade, the value is just discarded instead of breaking the upgrade completely.
```

Comment:
Are we sure this is a good idea? I mean, it seems like a bandaid for a bug we haven't found yet?

Reply:
1. Is this better or worse than not doing it?
2. What's the alternative?
Can we do anything to get to the root cause of the bug? E.g. crash with more info about the bug and ask people to send in the file. (which they likely can't from end user devices)
3. Are we only seeing this happening for old files? How old?
4. Can you reference bug reports?

Reply:
One example issue is this one: https://github.com/realm/realm-java/issues/7248. I will try to think it over once more how the size could end up being wrong. If the conclusion is that the size must be wrong in the file, then there is probably not much we can do except skip that value.

Reply:
Are there places where we can validate the string when we receive it? We have had some weird issues with strings before in Java since .. forever..